src/storage/core.rs

```diff
@@ -795,12 +795,24 @@ impl<K: Key> Storage<K> {
             .read()
             .await
             .iter()
-            .map(|blob| blob.check_filters(key.as_ref()))
-            .collect::<FuturesUnordered<_>>()
-            .any(|value| value)
-            .await;
+            .filter(|blob| !blob.is_filter_offloaded())
+            .any(|blob| blob.check_filters_in_memory(key.as_ref()));
 
-        Some(in_active || in_closed)
+        if !(in_active || in_closed) {
+            let in_closed_offloaded = inner
+                .blobs
+                .read()
+                .await
+                .iter()
+                .filter(|blob| blob.is_filter_offloaded())
```

Comment:
Is it possible that the bloom filter will be offloaded between the previous and current checks? If possible, the BLOB sets may diverge. In general, it would be safer to build a set of BLOBs with non-offloaded filters first, and then test the filters on BLOBs in that set and for blobs that are not in that set

Reply:
Done